sdks/python/apache_beam/transforms/util.py

```diff
@@ -634,12 +630,13 @@ def restore_timestamps(element, window=DoFn.WindowParam):
 
     ungrouped = pcoll | Map(reify_timestamps)
     ungrouped._windowing = Windowing(
-        window_fn,
+        window.GlobalWindows(),
         triggerfn=AfterCount(1),
         accumulation_mode=AccumulationMode.DISCARDING,
         timestamp_combiner=TimestampCombiner.OUTPUT_AT_EARLIEST)
     result = (ungrouped
               | GroupByKey()
+              | WindowInto(window_fn)
```

Comment:
seems line 641 is restoring the window, is applying window_fn here makes a difference?